exp/addons/internal/controllers/clusterresourcesetbinding_controller.go

```diff
@@ -77,8 +79,37 @@ func (r *ClusterResourceSetBindingReconciler) Reconcile(ctx context.Context, req
 		// Error reading the object - requeue the request.
 		return ctrl.Result{}, err
 	}
-
-	cluster, err := util.GetOwnerCluster(ctx, r.Client, binding.ObjectMeta)
+	// Before 1.4 cluster name was stored as ownerRef; this code migrated this value to
+	// the spec.clusterName field if necessary.
+	clusterName := binding.Spec.ClusterName
+	if clusterName == "" {
+		// Initialize the patch helper.
+		patchHelper, err := patch.NewHelper(binding, r.Client)
+		if err != nil {
+			log.Error(err, "Failed to configure the patch helper")
+			return ctrl.Result{}, err
+		}
+		// Update the clusterName field of the existing ClusterResourceSetBindings with ownerReferences.
+		// More details please refer to: https://github.com/kubernetes-sigs/cluster-api/issues/7669.
+		if clusterName, err = getClusterNameFromOwnerRef(binding.ObjectMeta); err != nil {
+			return ctrl.Result{}, err
+		} else if clusterName == "" {
+			log.Info("ownerRef not found for the ClusterResourceSetBinding")
+			return ctrl.Result{}, nil
+		}
```

Comment:
```suggestion
		}
```
We can drop this case now as it won't occur

Reply:
Thanks for the reminder~